### ISSUE TITLE
Mouse events are shifted after a css transform scale on the map container

### ISSUE THREAD

**Issue by nedasoo** [maplibre/maplibre-gl-js#1160]
Issue has been reported in MapboxGL github [here](https://github.com/mapbox/mapbox-gl-js/issues/6079) 

**maplibre-gl-js version**: 2.1.7

**browser**:  Google Chrome 100.0.4896.60

### Steps to Trigger Behavior

 1.Create a map
 2. Downscale the map container (Set `transform:scale(0.8)`)
 3. Use map.resize() to update the canvas.

### Link to Demonstration

Here is an example with a downscaled map (since the scale was set initially, no need for `map.resize()` )
https://jsfiddle.net/bjwtcqo6/

### Expected Behavior
All mouse events should match the updated map.

### Actual Behavior
All mouse events are shifted.

**Comment by DaveHeng** [comment]
solution:
```
.map {
    width: calc(100% * 0.8);
    height: calc(100% * 0.8);
    position: absolute;
    transform: translate(-50%, -50%) scale(calc(1 / 0.8));
    transform-origin: center center;
    top: 50%;
    left: 50%;
  }
  ```

**Comment by thehoneymad** [comment]
@HarelM, @louwers this is a blocker for AWS to migrate a service to MapLibre. :) Any chance we can prioritize this?

**Comment by louwers** [comment]
> @HarelM, @louwers this is a blocker for AWS to migrate a service to MapLibre. :) Any chance we can prioritize this?

I am maintainer for Native, not JS. In general I can say that only fixing cricical bugs is within the responsibilities of us maintainers. And since we do not manage a team it is hard for us to prioritize anything aside from calling attention to it.

**Comment by IvanSanchez** [comment]
@thehoneymad Sounds like you might want to convince @HarelM to assign a [bounty](https://github.com/maplibre/maplibre/wiki/Bounty-System) here.

**Comment by wipfli** [comment]
You are right @IvanSanchez that Bounties can be a good tool for fixing bug. However, we have to make sure that the work falls within one of the Bounty directions (https://github.com/maplibre/maplibre/issues?q=is%3Aissue+is%3Aopen+label%3A%22bounty+direction%22) which are approved by the MapLibre Governing Board. I am not sure if the ticket here is aligned with one of the Bounty directions, so at the moment we probably cannot assign a Bounty here.
